tools/create_basis_compressed_glbs.py

```diff
@@ -0,0 +1,253 @@
+import argparse
+import glob
+import itertools
+import json
+import multiprocessing
+import os
+import os.path as osp
+import shlex
+import shutil
+import subprocess
+from typing import Any, Callable, List, Optional, Tuple
+
+import tqdm
+
+TOOL_PATH = osp.realpath(
+    osp.join(
+        osp.dirname(__file__),
+        "..",
+        "src/deps/magnum-plugins/src/MagnumPlugins/TinyGltfImporter/Test",
+    )
+)
+
+IMAGE_CONVERTER = osp.realpath(
+    osp.join(
+        osp.dirname(__file__), "..", "build/utils/imageconverter/magnum-imageconverter"
+    )
+)
+
+
+def build_parser(
+    parser: Optional[argparse.ArgumentParser] = None,
+) -> argparse.ArgumentParser:
+    if parser is None:
+        parser = argparse.ArgumentParser(
+            description="Tool to convert all glbs in a given directory to basis compressed glbs."
+            "  This allows them to be loaded faster onto the GPU and use between 4x and 6x less GPU memory.",
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        )
+
+    parser.add_argument(
+        "--num-workers",
+        default=2,
+        type=int,
+        help="Number of workers to run in parallel. Most of the compression process is multi-threaded and set to use 4 threads.  Budget around ~3 CPU cores per worker.",
+    )
+    parser.add_argument(
+        "--basedir", type=str, required=True, help="Base directory to find GLBs in."
+    )
+    parser.add_argument(
+        "--pre-clean",
+        action="store_true",
+        help="Clean artifacts before running the script",
+    )
+    parser.add_argument(
+        "--rename-basis",
+        action="store_true",
+        help="Rename the basis meshes to *.glb, and original glbs to *.glb.orig.  Otherwise the basis meshes are named *.basis.glb",
+    )
+    parser.add_argument(
+        "--niceness", type=int, default=10, help="Niceness value for all the workers"
+    )
+    parser.add_argument(
+        "--convert-unlit",
+        action="store_true",
+        help="Convert meshes to unlit.  This is useful when processing reconstructions of real scenes or objects that have 'baked' lighting as it signifies to habitat that this object should use flat shading.",
+    )
+
+    return parser
+
+
+def extract_images(mesh_name: str) -> None:
+    output_dir = osp.splitext(mesh_name)[0] + "_hab_basis_tool"
+    os.makedirs(output_dir, exist_ok=True)
+
+    tool = osp.join(TOOL_PATH, "glb2gltf.py")
+    mesh_name = osp.abspath(mesh_name)
+    output_name = (
+        osp.splitext(osp.join(output_dir, osp.basename(mesh_name)))[0] + ".gltf"
+    )
+
+    subprocess.check_output(
+        shlex.split(f"{tool} {mesh_name} --extract-images --output {output_name}"),
+        cwd=output_dir,
+    )
+
+
+def img_name_to_basis(img: str) -> str:
+    return osp.splitext(img)[0] + ".basis"
+
+
+def convert_image_to_basis(img: str) -> None:
+    basis_img = img_name_to_basis(img)
+
+    settings = "threads=4,mip_gen=true,compression_level=2"
```

Comment:
What's `compression_level`? Can we document here? Is this an interesting choice for the user?

Reply:
Basis doesn't document it with anything more descriptive than "the overall effort the compressor uses on a 1-5 scale".  I'll expose it, but there isn't a great rule of thumb for setting it when you can't manually verify if there are compression artifacts (higher may reduce them).  I just put its as high as I could while still keeping things to a reasonable amount of time and checked a few meshes for artifacts.